dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -2187,16 +2189,16 @@ DomainParticipantImpl::handle_exception(ACE_HANDLE /*fd*/)
       = pubPtr->delete_contained_entities();
 
       if (result != DDS::RETCODE_OK) {
-        return result;
+        ret = result;
```

Comment:
Before this change, a non-OK result stopped further processing.  Now it's proceeding with the next steps.  Is that what we want?